### DISCUSSION TITLE
Sunbird release version (tags) - release-4.10.0_RC12 : Forbidden: API WHITELIST Access is denied

### DISCUSSION THREAD

**Discussion by sanjeev-scm** [Sunbird-Lern/Community#157] [Issues] [answered]
Hi Team,

Sunbird release version (tags) - release-4.10.0_RC12

We've created a new API for the Sunbird ED portal.
The API is written in python and is not a micro-service.
The API is deployed on the VM.

The API URL is:
{{host}}/learner/course/v1/progress/reports/do_113876534913794048133/0138765499798814726?limit=200&offset=0

We have also whitelisted the API in as follows:
  '/learner/course/v1/progress/reports/:courseId/:batchId': {
   checksNeeded: ['ROLE_CHECK'],
   ROLE_CHECK: [
    ROLE.CONTENT_CREATOR,
    ROLE.COURSE_CREATOR,
    ROLE.BOOK_CREATOR,
    ROLE.ORG_ADMIN,
    ROLE.ADMIN,
   ]
  },

We are able to access the API in postman by prefixing /api before learner in the URL
However, when we try to access the API in the portal, we get the following error.

403 Forbidden error with the response as follows.

Response:
{
  "id": "api.error",
  "ver": "1.0",
  "ts": "2023-08-30 12:15:27:621+0000",
  "params": {
    "resmsgid": "e7b01b50-472e-11ee-9bba-0f5519ab4c68",
    "msgid": null,
    "status": "failed",
    "err": "FORBIDDEN_ERROR",
    "errmsg": "Forbidden: API WHITELIST Access is denied"
  },
  "responseCode": "FORBIDDEN",
  "result": {}
}

Request you to assist us in resolving this issue.

**Comment by AmiableAnil** [comment]
@gandham-santhosh Can you please help here?

**Reply by gandham-santhosh** [reply]
@sanjeev-scm which service is giving 403? If you are getting 403 from APIManager/kong then there is some issue with onboarding the API. Otherwise check if the whitelisting of the API at portal is added correctly

**Comment by mohitga** [comment]
@sanjeev-scm can you please share details on which service is throwing 403 error?

**Reply by sanjeev-scm** [reply, marked as answer]
Hi Mohit and Santhosh

We have whitelisted the api at portal as suggested by Santosh Gandham and this issue is resolved now.
Request you to mark it as closed.

**Reply by mohitga** [reply]
Hi @sanjeev-scm  
Great, can you mark the thread answered, we will take care after it.

**Reply by sanjeev-scm** [reply]
I have marked the answer
